streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/DelayProcessor.kt

```diff
@@ -25,20 +27,41 @@ class DelayProcessor<T>(
         private val timerFactory: TimerFactory
     ) : ProcessorSubscription<T, T>(subscriber) {
 
+        private val serialQueue = SynchronousSerialQueue()
         private val cancellableManagerProvider = CancellableManager()
 
+        private var blocksToRun: List<() -> Unit> by atomic(emptyList())
```

Comment:
Maybe we could use AtomicListReference and we can use it as a regular mutable list.

Reply:
Good suggestion I din't know this existed. 👍🏻 

Howerver, I tried using it but there's a few missing operators on that AtomicListReference, such as `.clear()` and `.drop(n)` for example, which would make it easier to read.

For `.clear()` I could use `blocksToRun.removeAll(blocksToRun.value)`, but this does not seem too much eficient. Same thing for `.drop(n)` where I could do the following :

```
blocksToRun.value.firstOrNull()?.let { block ->
    block.invoke()
    blocksToRun.remove(block)
}
```

What do you think about keeping the `atomic(emptyList())` and updating it aftwerards once we implement these operators on the AtomicListReference?

Reply:
@npresseault Merging this for now as we need a quick (and clean) aternative to the latest `.debounce()` breaking change. I'll address the AtomicListReference issue later if needed. Thanks

Reply:
Yes, works for me. Didn't notice not everything was implemented with AtomicListReference